cranelift/codegen/src/machinst/abi.rs

```diff
@@ -2453,14 +2400,15 @@ impl<M: ABIMachineSpec> CallSite<M> {
         ctx.abi_mut()
             .accumulate_outgoing_args_size(ret_space + arg_space);
 
-        // Any adjustment to SP to account for required outgoing arguments/stack return values must
-        // be done around the call, to ensure that SP is always in a consistent state for all other
-        // writes.
-        for inst in M::gen_reserve_argument_area(ret_space + arg_space) {
-            ctx.emit(inst);
-        }
-
         let tmp = ctx.alloc_tmp(word_type).only_reg().unwrap();
+
+        // Any adjustment to SP to account for required outgoing arguments/stack return values must
+        // be done inside of the the call pseudo-op, to ensure that SP is always in a consistent
+        // state for all other instructions. For example, if a tail-call abi function is called
+        // here, the reclamation of the outgoing argument area must be done inside of the call
+        // pseudo-op's emission to ensure that SP is consisten at all other points in the lowered
+        // function. (Except the prologue and epilogue, but those are fairly special parts of the
+        // function that establish the SP invariants that are relied on elsewhere.)
```

Comment:
(amend comment) ...and are generated after the register allocator has run and thus cannot have register allocator-inserted references to SP offsets.